pkg/authentication/basic/htpasswd.go

```diff
@@ -30,59 +33,114 @@ type sha1Pass string
 // NewHTPasswdValidator constructs an httpasswd based validator from the file
 // at the path given.
 func NewHTPasswdValidator(path string) (Validator, error) {
+	h := &htpasswdMap{users: make(map[string]interface{})}
+
+	err := h.loadHTPasswdFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = watcher.WatchFileForUpdates(path, nil, func() {
+		err := h.loadHTPasswdFile(path)
+		if err != nil {
+			logger.Errorf("%v: no changes were made to the current htpasswd map", err)
+		}
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Please wrap with context

Reply:
The context in already in the `oauthproxy.go` file -- same as above.

Reply:
Wrapped the error with context